### ISSUE TITLE
[Rancher] RKE2 - AWS Cloud Provider not working

### ISSUE THREAD

**Issue by kinarashah** [rancher/rancher#34233]
RKE2 provisioned clusters aren't working with AWS cloud provider. ec2 instances are missing the tags https://rancher.com/docs/rke/latest/en/config-options/cloud-providers/aws/#tagging-aws-resources 

Original issue tracking RKE2 kubelet and KDM fix: 
https://github.com/rancher/rancher/issues/34105

**Comment by kinarashah** [comment]
Note: This issue can be tested by itself, bring up an ec2 cluster and validate instances get the cluster tags. But the actual AWS cloud provider issue is dependent on #34105 to be moved to test.

**Comment by kinarashah** [comment]
Available to test with `v1.21.3-rc6+rke2r2`.

**Comment by davidnuzik** [comment]
Related: https://github.com/rancher/rancher/issues/34105#issuecomment-900652972 (aws cloud provider now working).


 
**One check FAILED, but we're going to call this overall a pass because the intended change here works**
 
**Reproduction Steps:**

Not required.
 
**Validation Steps:**

Rancher version: v2.6-head `72986b2e4` pulled at 8/17/21 1:32 pm Pacific
Rancher cluster type: single-node docker
Docker version: 20.10
 
Downstream cluster type: rke2
Downstream K8s version: v1.21.3-rc6+rke2r2
 
Validation steps:

1. Create an RKE2 AWS cluster in rancher. Chose "Amazon" as the cloud provider and provide a suitable IAM role.
2. When the cluster is up, check the ec2 instance(s). They have the expected tag as per [the documentation](https://rancher.com/docs/rke/latest/en/config-options/cloud-providers/aws/#tagging-aws-resources).
![image](https://user-images.githubusercontent.com/45179589/129805380-ad961f0e-ede5-43a5-834a-b5137c90d63b.png)
3. Create a Load Balancer, confirmed it indeed gets created in aws web console with correct tag.
![image](https://user-images.githubusercontent.com/45179589/129807032-c9c25f1b-defb-4965-b6e7-b0f4a09df7a3.png)
_Note my clusterid is different in this image, this is okay I was testing with a different cluster with more complexity, etc. Also, I believe this change was ONLY for ec2 tags so suspect the tags were working prior for lb and storage volumes -- I checked this anyway since it's related._
 
4. Delete the cluster. The storage volume and load balancer are NOT cleaned up. (But ec2 instances are, as expected)

_I am pretty confident that the issue of these objects not being cleaned up is a separate issue._
I also checked with rke1 and it too does not clean up storage volumes/lb upon cluster deletion (just ec2 instances). But it looks like this is how things have been behaving in the past - even in v2.5.9. I created a separate enhancement request here for Rancher to clean up these objects upon cluster creation: https://github.com/rancher/rancher/issues/34293



Additional Info:
While testing, I ensured UI request payloads appeared correct.
Further testing of aws cloud provider will be done via https://github.com/rancher/rancher/issues/32390
